appBankAutomation.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+)
+
+type PretixBankAutomation struct {
+	// filtered
+}
+
+func (e PretixBankAutomation) Init() {
+
+}
+
+func (e PretixBankAutomation) Run() {
+	// 1. Get all transactions from the last 24 hours
+	transactions, err := getTransactionsFromLast24Hours()
+	if err != nil {
+		msg := fmt.Sprintf("Error getting transactions: %v", err)
+		log.Fatalf(msg)
+		sendEmailNotification(msg)
+
+	}
+
+	// 2. Scan the remittanceInformationUnstructured for the keyword {{EVENT_SLUG}}{{ORDER_CODE}}
+
+	for _, transaction := range transactions {
+
+		if transaction.DebtorAccount.IBAN == "" {
+			// check if deposit or withdrawal
+			continue
+		}
+		if transaction.RemittanceInformationUnstructured == "" {
+			// check if remittance empty
+			continue
+		}
+		//DEBUG
+		transaction.RemittanceInformationUnstructured = "FEST-F7CXX "
+		//DEBUG
+		bankAutomationError.RemittanceInformation = transaction.RemittanceInformationUnstructured
+		bankAutomationError.FromAccount = transaction.DebtorAccount.IBAN
+
+		orderCode, err := parseRemittanceInformation(transaction.RemittanceInformationUnstructured, pretixConfig.EventSlug)
+		if err != nil {
+			addBankAutomationError(err.Error())
+			log.Printf("%v RemittanceInfo: %s", err, transaction.RemittanceInformationUnstructured)
+			continue
+		}
+		bankAutomationError.Code = orderCode
+
+		// 3. Get order from Pretix using orderCode
+		order, err := getPretixOrder(orderCode)
+		if err != nil {
+			addBankAutomationError(err.Error())
+			log.Printf("%v OrderCode: %s", err, orderCode)
```

Comment:
Minor nitpick, `Printf` doesn't add a newline. You might want to use `Println` for these or add it manually with `\n`

Reply:
changed all printf with println